Dockerfile

```diff
@@ -1,56 +1,217 @@
+        php5.6-json \
+        php5.6-mbstring \
+        php5.6-phar \
+        php5.6-phpdbg \
+        php5.6-readline \
+        php5.6-sockets \
+        php5.6-xml \
+        php5.6-xsl \
+        php5.6-zip \
+        \
+        php7.0-cli \
+        php7.0-bz2 \
+        php7.0-curl \
+        php7.0-dev \
+        php7.0-fileinfo \
+        php7.0-intl \
+        php7.0-json \
+        php7.0-mbstring \
+        php7.0-phar \
+        php7.0-phpdbg \
+        php7.0-readline \
+        php7.0-sockets \
+        php7.0-xml \
+        php7.0-xsl \
+        php7.0-zip \
+        \
+        php7.1-cli \
+        php7.1-bz2 \
+        php7.1-curl \
+        php7.1-dev \
+        php7.1-fileinfo \
+        php7.1-intl \
+        php7.1-json \
+        php7.1-mbstring \
+        php7.1-phar \
+        php7.1-phpdbg \
+        php7.1-readline \
+        php7.1-sockets \
+        php7.1-xml \
+        php7.1-xsl \
+        php7.1-zip \
+        \
+        php7.2-cli \
+        php7.2-bz2 \
+        php7.2-curl \
+        php7.2-dev \
+        php7.2-fileinfo \
+        php7.2-intl \
+        php7.2-json \
+        php7.2-mbstring \
+        php7.2-phar \
+        php7.2-phpdbg \
+        php7.2-readline \
+        php7.2-sockets \
+        php7.2-xml \
+        php7.2-xsl \
+        php7.2-zip \
+        \
+        php7.3-cli \
+        php7.3-bz2 \
+        php7.3-curl \
+        php7.3-dev \
+        php7.3-fileinfo \
+        php7.3-intl \
+        php7.3-json \
+        php7.3-mbstring \
+        php7.3-phar \
+        php7.3-phpdbg \
+        php7.3-readline \
+        php7.3-sockets \
+        php7.3-xml \
+        php7.3-xsl \
+        php7.3-zip \
+        \
+        php7.4-cli \
+        php7.4-bz2 \
+        php7.4-curl \
+        php7.4-dev \
+        php7.4-fileinfo \
+        php7.4-intl \
+        php7.4-json \
+        php7.4-mbstring \
+        php7.4-phar \
+        php7.4-phpdbg \
+        php7.4-readline \
+        php7.4-sockets \
+        php7.4-xml \
+        php7.4-xsl \
+        php7.4-zip \
+        \
+        php8.0-cli \
+        php8.0-bz2 \
+        php8.0-curl \
+        php8.0-dev \
+        php8.0-fileinfo \
+        php8.0-intl \
+        php8.0-mbstring \
+        php8.0-phar \
+        php8.0-phpdbg \
+        php8.0-readline \
+        php8.0-sockets \
+        php8.0-xml \
+        php8.0-xsl \
+        php8.0-zip \
+        \
+        php8.1-cli \
+        php8.1-bz2 \
+        php8.1-curl \
+        php8.1-dev \
+        php8.1-fileinfo \
+        php8.1-intl \
+        php8.1-mbstring \
+        php8.1-phar \
+        php8.1-phpdbg \
+        php8.1-readline \
+        php8.1-sockets \
+        php8.1-xml \
+        php8.1-xsl \
+        php8.1-zip \
+    # Set default PHP version
+    && update-alternatives --set php /usr/bin/php7.4 \
+    && update-alternatives --set phpize /usr/bin/phpize7.4 \
+    && update-alternatives --set php-config /usr/bin/php-config7.4 \
+    && apt autoremove -y \
```

Comment:
TBH, I found the split setup scripts just much harder to work with: changing the repo can always be done, but practically, having `apt` operate once leads to a smaller build time, and a clearer list of dependencies at one glance.

Reply:
Sorry, rephrasing to better explain why the previous setup was just harder to follow:

I needed to browse a number of files to figure out what is installed, and how, instead of having a single, clearly defined list.

I would expect bash scripts only where hacks are required.

Reply:
You can also provide `--squash` to docker build if you want to reduce the final image size? It's experimental still (but has been for at least a year now from what I can remember) so I'm not certain this is an option, but might be worth checking?

Reply:
With the current image format, that would be ~3% image reduction, so IMO not sure if worth doing (at the cost of reducing image download parallelization).

Unsure of the effect of squashing against buildx caches 🤔 

In practice, even when squashing, I'd still rather have all the `apt` dependencies explicitly listed in `Dockerfile`, than hidden somewhere :D

Reply:
Ah OK. It works better if there are files in an intermediate step which aren't in the final image, which I guess isn't the case here.

Reply:
I will 🚢  this one as-is for now: let's discuss over potential improvements (or rollback) of this particular layer in a separate issue, if it is a problem, but overall, I need this to be fixed for CI to run again in downstream projects.

Note that this also opens the door for installing more composer-based tooling in the near future 👍